main.go

```diff
@@ -272,3 +272,22 @@ func startPollServer(out chan<- *net.IP, localIp *net.IP) {
 		}
 	}()
 }
+
+func readSecret(envName string) string {
+	secret := os.Getenv(envName)
+
+	if secret != "" {
+		return secret
+	}
+
+	passwordFilePath := os.Getenv(envName + "_FILE")
+	if passwordFilePath != "" {
+		content, err := os.ReadFile(passwordFilePath)
+		if err != nil {
+			slog.Error("Failed to read secret from file "+passwordFilePath, logging.ErrorAttr(err))
+		} else {
+			secret = string(content)
```

Comment:
```suggestion
			secret = strings.TrimSuffix(strings.TrimSuffix(string(content), "\r\n"), "\n")
```
Some ways to put the secret into the file (echo without `-n`, nano without `-L`, probably more), leave a newline at the end, this should strip it to not cause any unexplained errors.